src/test/java/hudson/plugins/git/GitSCMTest.java

```diff
@@ -2738,6 +2738,38 @@ public void testCommitMessageIsPrintedToLogs() throws Exception {
         assertThat(values, hasItem("Commit message: \"test commit\""));
     }
 
+    @Issue("JENKINS-49757")
+    @Test
+    public void testRedundantFetchCallFromLogs() throws Exception {
```

Comment:
Instead of using the getLogs() logic, I have taken the liberty to copy-paste the methodology used in [testCommitMessageIsPrintedToLogs()](https://github.com/jenkinsci/git-plugin/blob/23158ea1289900a4bfda8f3275eb9ff97d4346a5/src/test/java/hudson/plugins/git/GitSCMTest.java#L2722-L2738) 

I was able to retrieve the build logs from the freestyle project and count the number of "git-fetch" args in the log.